samples/openthread/coprocessor/src/rcp_hci.c

```diff
@@ -0,0 +1,384 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#if defined(CONFIG_RCP_SAMPLE_HCI)
```

Comment:
nit: I think this check in `.c` file might be redundant as we are only adding the file to the target if:
`target_sources_ifdef(CONFIG_RCP_SAMPLE_HCI app PRIVATE src/rcp_hci.c)`